extensions/amp-smartlinks/0.1/linkmate.js

```diff
@@ -177,16 +177,19 @@ export class Linkmate {
    * @public
    */
   mapLinks_() {
-    return this.linkmateResponse_.map(smartLink => {
-      return Array.prototype.slice.call(this.anchorList_)
-          .map(anchor => {
-            return {
-              anchor,
-              replacementUrl: anchor[this.linkAttribute_] === smartLink['url']
-                && smartLink['auction_id']
-                ? `https://shop-links.co/${smartLink['auction_id']}/?amp=true` : null,
-            };
-          });
-    })[0];
+    const mappings = [];
+    this.anchorList_.forEach(anchor => {
+      let replacementUrl = null;
+
+      this.linkmateResponse_.forEach(smartLink => {
+        if (anchor[this.linkAttribute_] === smartLink['url']
+            && smartLink['auction_id']) {
+          replacementUrl = `https://shop-links.co/${smartLink['auction_id']}/?amp=true`;
+        }
+      });
+      mappings.push({anchor, replacementUrl});
```

Comment:
do you need to check if replacementUrl is null?

Reply:
Nope. We want it to be null for a link we don't monetize. That way the replacement list knows it's not ours to monetize. 

I can double check the link replacement code and see if we can just omit nulls from this list.